core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1259,15 +1255,31 @@ public RelBuilder projectPlus(RexNode... nodes) {
     return projectPlus(ImmutableList.copyOf(nodes));
   }
 
-  /** Creates a {@link Project} of all original fields, plus the given list of
+  /** Creates a {@link Project} of all original fields, plus the given
    * expressions. */
   public RelBuilder projectPlus(Iterable<RexNode> nodes) {
     final ImmutableList.Builder<RexNode> builder = ImmutableList.builder();
     return project(builder.addAll(fields()).addAll(nodes).build());
   }
 
-  /** Creates a {@link Project} of the given list
-   * of expressions, using the given names.
+  /** Creates a {@link Project} of all original fields, minus the given
+   * expressions. */
+  public RelBuilder projectMinus(RexNode... expressions) {
+    return projectMinus(ImmutableList.copyOf(expressions));
+  }
+
+  /** Creates a {@link Project} of all original fields, minus the given
+   * expressions. */
+  public RelBuilder projectMinus(Iterable<RexNode> expressions) {
+    List<RexNode> allExpressions = new ArrayList<>(fields());
```

Comment:
Thanks @zabetak 
IIUC, when one in `expressions` matches multiple ones in `allExpressions`, there's only one item in `allExpressions` will be removed. If we really mean it, shall we doc and have a test ?

Reply:
I think this cannot happen because fields() are unique. Isn't this the case?

Reply:
True, you are right ~

Reply:
Since it's only valid to pass `expressions` as a list of `RexInputRef`. How about:
1. We check the type of `expressions` -- only `RexInputRef` is valid
2. Indexes of `RexInputRef` from `expressions` should be unique and smaller than `fields().size`

By above two steps, I think we can save the exception below.

Reply:
In which exception are you referring to? Aren't we gonna need at least one?

Reply:
I'm referring to `RESOURCE.expressionNotFound(excludeExp.toString())`. 
Yes we need at least one when duplicates